browser/ui/views/split_view/split_view_separator.cc

```diff
@@ -58,14 +65,61 @@ class MenuButtonDelegate : public views::WidgetDelegateView {
 
     image_button->SetImageHorizontalAlignment(views::ImageButton::ALIGN_CENTER);
     image_button->SetImageVerticalAlignment(views::ImageButton::ALIGN_MIDDLE);
+
+    background_animation_.SetSlideDuration(base::Milliseconds(150));
+
+    SetNotifyEnterExitOnChild(true);
   }
 
   ~MenuButtonDelegate() override = default;
 
+  // views::WidgetDelegateView:
+  void OnMouseEntered(const ui::MouseEvent& event) override {
+    background_animation_.Show();
+  }
+
+  void OnMouseExited(const ui::MouseEvent& event) override {
+    background_animation_.Hide();
+  }
+
+  void OnPaintBackground(gfx::Canvas* canvas) override {
+    auto scoped_canvas = TransformCanvasForBackground(canvas);
+    views::WidgetDelegateView::OnPaintBackground(canvas);
+  }
+
+  void OnPaintBorder(gfx::Canvas* canvas) override {
+    auto scoped_canvas = TransformCanvasForBackground(canvas);
+    views::WidgetDelegateView::OnPaintBorder(canvas);
+  }
```

Comment:
In order to keep button area big enough, we only transform the paiting instead of sizing the widget or button.